File: variant-v/java-service/src/main/java/com/flashsale/dto/OrderDtos.java
package com.flashsale.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

public class OrderDtos {
    
    @Data
    public static class OrderCreateRequest {
        private String customerEmail;
        private String skuId;
        private Integer quantity;
        private BigDecimal unitPrice;
        private String flashSaleCampaignId;
    }
    
    @Data
    public static class OrderCreateResponse {
        private String auditId;
        private String orderId;
        private String status;
        private String message;
    }
    
    @Data
    public static class OrderStatusResponse {
        private String auditId;
        private String status;
        private LocalDateTime createdAt;
    }
}

File: variant-v/csharp-service/DTOs/OrderDtos.cs
namespace FlashSale.Api.V.DTOs;

public class OrderDtos
{
    public class OrderCreateRequest
    {
        public string CustomerEmail { get; set; } = string.Empty;
        public string? SkuId { get; set; }
        public int Quantity { get; set; }
        public decimal UnitPrice { get; set; }
        public string? FlashSaleCampaignId { get; set; }
    }

    public class OrderCreateResponse
    {
        public Guid AuditId { get; set; }
        public Guid OrderId { get; set; }
        public string Status { get; set; } = string.Empty;
        public string Message { get; set; } = string.Empty;
    }

    public class OrderStatusResponse
    {
        public Guid AuditId { get; set; }
        public string Status { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; }
    }
}
